app/controllers/rooms_controller.rb

```diff
@@ -16,11 +16,13 @@ def index
       @buildings = Building.where(bldrecnbr: buildings_ids, visible: false).order(:name)
     else
       @buildings = Building.where(bldrecnbr: buildings_ids).order(:name)
+      @buildings = Building.all.order(:name)
```

Comment:
Line 19 is changing the collection of buildings to all buildings. This is not correct. We show only buildings that have classrooms. This is why we need the buidings_ids variable, it has the ids of buildings with classrooms.